jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java

```diff
@@ -106,8 +139,23 @@ TargetImage getTargetImage() {
     return Optional.ofNullable(executorService);
   }
 
-  Optional<CacheConfiguration> getCacheConfiguration() {
-    return Optional.ofNullable(cacheConfiguration);
+  Path getBaseImageLayersCacheDirectory() {
+    return baseImageLayersCacheDirectory;
+  }
+
+  Path getApplicationLayersCacheDirectory() throws CacheDirectoryCreationException {
```

Comment:
Who's calling this? Can we guarantee this is not called concurrently?

Reply:
This is to-be-called by `JibContainerBuilder#containerize`. None of the API classes are thread-safe and should not be called concurrently.